scipy/special/tests/test_loggamma.py

```diff
@@ -35,23 +35,34 @@ def f(z):
     FuncData(f, dataset, 0, 1, rtol=1e-14, atol=1e-14).check()
 
 
-def test_realpart():
+def test_complex_dispatch_realpart():
     # Test that the real parts of loggamma and gammaln agree on the
     # real axis.
     x = np.r_[-np.logspace(10, -10), np.logspace(-10, 10)] + 0.5
+
     dataset = np.vstack((x, gammaln(x))).T
 
     def f(z):
+        z = np.array(z, dtype='complex128')
         return loggamma(z).real
-    
+
     FuncData(f, dataset, 0, 1, rtol=1e-14, atol=1e-14).check()
 
 
+def test_real_dispatch():
+    x = np.logspace(-10, 10) + 0.5
+    dataset = np.vstack((x, gammaln(x))).T
+
+    FuncData(loggamma, dataset, 0, 1, rtol=1e-14, atol=1e-14).check()
+    assert_(loggamma(0) == np.inf)
+    assert_(np.isnan(loggamma(-1)))
```

Comment:
Doesn't pytest support `assert`?

Reply:
Yes... but I seem to recall people not liking that. Can't find the relevant discussions now though (if they ever existed). @rgommers was there such a thing or am I just making stuff up?

Reply:
Never really been discussed, but yes ``pytest`` seems to modify the behavior of ``python -O`` somehow. I wouldn't like to rely on something ``pytest`` specific for this. And I suspect whatever they did there is fairly hacky.